docker/Dockerfile

```diff
@@ -1,33 +1,225 @@
+    add-apt-repository ppa:deadsnakes/ppa && \
+    apt-get update && \
+    DEBIAN_FRONTEND=noninteractive $APT_INSTALL \
+        python3.6 \
+        python3.6-dev \
+        python3.6-tk \
+        python3-distutils \
+        cython \
+        libopenmpi-dev \
+        openmpi-bin \
+        libsm6 \
+        libxext6 \
+        libxrender-dev \
+        locales \
+        mesa-utils
+
+# ==================================================================
+# Python goodies
+# ------------------------------------------------------------------
+RUN APT_INSTALL="apt-get install -y --no-install-recommends" && \
+    PIP_INSTALL="python -m pip --no-cache-dir install --upgrade" && \
+    GIT_CLONE="git clone --depth 10" && \
+    rm -rf /var/lib/apt/lists/* \
+           /etc/apt/sources.list.d/cuda.list \
+           /etc/apt/sources.list.d/nvidia-ml.list && \
+    apt-get update && \
+    wget -O ~/get-pip.py \
+        https://bootstrap.pypa.io/get-pip.py && \
+    python3.6 ~/get-pip.py && \
+    ln -s /usr/bin/python3.6 /usr/local/bin/python3 && \
+    ln -s /usr/bin/python3.6 /usr/local/bin/python && \
+    $PIP_INSTALL \
+        setuptools \
+        && \
+    $PIP_INSTALL \
+        cloudpickle==1.4.1 \
+        cmake==3.15.3 \
+        Cython==0.29.13 \
+        deepdiff==4.3.2 \
+        defusedxml==0.6.0 \
+        flake8==3.7.8 \
+        glances==3.1.4.1 \
+        h5py==2.10.0 \
+        imagesize==1.1.0 \
+        ipdb==0.13.2 \
+        jupyter==1.0.0 \
+        magicattr==0.1.4 \
+        matplotlib==3.1.1 \
+        numpy==1.17.2 \
+        opencv-python==4.1.1.26 \
+        ordered-set==3.1.1 \
+        pandas==0.25.1 \
+        Pillow==6.1.0 \
+        protobuf==3.9.2 \
+        pycodestyle==2.5.0 \
+        pyflakes==2.1.1 \
+        ray==0.8.5 \
+        scipy==1.3.1 \
+        six==1.12.0 \
+        scikit-learn==0.20.1 \
+        scikit-optimize==0.7.4 \
+        tabulate==0.8.7 \
+        tensorboard==1.14.0 \
+        tensorflow==1.14.0 \
+        tensorboardX==2.0 \
+        threadpoolctl==2.0.0 \
+        torch==1.5.0 \
+        torchvision==0.6.0 \
+        tqdm==4.43.0
+
+# ==================================================================
+# cuDNN
+# ==================================================================
+RUN APT_INSTALL="apt-get install -y --no-install-recommends" && \
+    rm -rf /var/lib/apt/lists/* \
+           /etc/apt/sources.list.d/cuda.list \
+           /etc/apt/sources.list.d/nvidia-ml.list && \
+    apt-get update && \
+    mkdir -p /opt/cudnn7 && \
+    cd /opt/cudnn7/ && \
+    wget http://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64/nvidia-machine-learning-repo-ubuntu1804_1.0.0-1_amd64.deb && \
+    dpkg -i nvidia-machine-learning-repo-ubuntu1804_1.0.0-1_amd64.deb && \
+    apt-get update && \
+    $APT_INSTALL libcudnn7-dev && \
+    ldconfig && \
+    apt-get clean && \
+    apt-get autoremove && \
+    rm -rf /var/lib/apt/lists/* /tmp/* ~/*
```

Comment:
Unneeded, just use the cudnn runtime from nvidia like: FROM nvidia/cuda:10.2-cudnn7-runtime-ubuntu18.04

Reply:
We may want to consider `FROM nvidia/cudagl:10.1-devel-ubuntu18.04`, as this allows users to invoke and view OpenGL-based rendering applications from within the Docker container.

I'm not aware of a base that supports OpenGL-based rendering + also has cuDNN.